sections/exchange/TradeCard/Cards/QuoteCurrencyCard.tsx

```diff
@@ -1,36 +1,60 @@
-import { FC, memo, useCallback } from 'react';
+import { FC, memo, useCallback, useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
-import { useRecoilValue } from 'recoil';
+import { updateBaseAmount } from 'state/exchange/actions';
+import { setQuoteAmount } from 'state/exchange/reducer';
+import { setMaxQuoteBalance, setOpenModal } from 'state/exchange/reducer';
+import {
+	selectQuoteBalanceWei,
+	selectQuoteCurrencyName,
+	selectQuotePriceRateWei,
+} from 'state/exchange/selectors';
+import { useAppDispatch, useAppSelector } from 'state/hooks';
 
-import { useExchangeContext } from 'contexts/ExchangeContext';
-import { quoteCurrencyKeyState, quoteCurrencyAmountState } from 'store/exchange';
+import useDebouncedMemo from 'hooks/useDebouncedMemo';
 
 import CurrencyCard from '../CurrencyCard';
 
 const QuoteCurrencyCard: FC = memo(() => {
 	const { t } = useTranslation();
-	const quoteCurrencyKey = useRecoilValue(quoteCurrencyKeyState);
-	const quoteCurrencyAmount = useRecoilValue(quoteCurrencyAmountState);
 
-	const {
-		quoteCurrencyBalance,
-		setOpenModal,
-		allTokensMap,
-		quotePriceRate,
-		onQuoteCurrencyAmountChange,
-		onQuoteBalanceClick,
-	} = useExchangeContext();
+	const { quoteCurrencyKey, quoteAmount } = useAppSelector(({ exchange }) => ({
+		quoteCurrencyKey: exchange.quoteCurrencyKey,
+		quoteAmount: exchange.quoteAmount,
+	}));
 
-	const openQuoteModal = useCallback(() => setOpenModal('quote-select'), [setOpenModal]);
+	const quoteBalance = useAppSelector(selectQuoteBalanceWei);
+	const quotePriceRate = useAppSelector(selectQuotePriceRateWei);
+	const quoteCurrencyName = useAppSelector(selectQuoteCurrencyName);
+
+	const dispatch = useAppDispatch();
+
+	const quoteAmountDebounced = useDebouncedMemo(() => quoteAmount, [quoteAmount], 300);
```

Comment:
Another place where it seems like a lot of logic is duplicated between mobile and desktop, would be good to consolidate, maybe we can add a TODO